### PR TITLE
ddl: make ddl session be a common function

### LINKED ISSUE #40709: make session be a common function
## Enhancement
This issue will refactor the session code to make it be a common function and can be used by different module.

### PR DESCRIPTION
**Pull request by Benjamin2037**
### What problem does this PR solve?
This is a refactor pr to make session be a common function.

Issue Number: close #40709

Problem Summary:

### What is changed and how it works?
Move the session implement code into util/sessionutil and make it public usage.

### Check List

Tests 

- [X] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: util/sessionutil/session_test.go [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+	store, _ := testkit.CreateMockStoreAndDomain(t)
+	s := testkit.NewTestKit(t, store).Session()
+	se := su.NewSession(s)
+	se.Begin()
+	_, err := se.GetTxn()
+	requires.NoError(t, err)
+}
+
+func TestSession(t *testing.T) {
+	store, _ := testkit.CreateMockStoreAndDomain(t)
```
Comment (hawkingrei) [inline util/sessionutil/session_test.go:60]:
```suggestion
	store := testkit.CreateMockStore(t)
```

File: util/sessionutil/session_test.go [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+	s := testkit.NewTestKit(t, store).Session()
+	se := su.NewSession(s)
+	se.Begin()
+	s1 := se.Session()
+	requires.Equal(t, s1, s)
+}
+
+func TestRunInTxn(t *testing.T) {
+	store, _ := testkit.CreateMockStoreAndDomain(t)
```
Comment (hawkingrei) [inline util/sessionutil/session_test.go:69]: ditto

File: ddl/index.go [hunk 3]
```diff
@@ -747,10 +748,10 @@ func canUseIngest(w *worker) bool {
 
 // IngestJobsNotExisted checks the ddl about `add index` with ingest method not existed.
 func IngestJobsNotExisted(ctx sessionctx.Context) bool {
-	sess := session{ctx}
+	sess := su.Session{Context: ctx}
```
Comment (hawkingrei) [inline ddl/index.go:751]:
```suggestion
	sess := su.NewSession(ctx)
```

**Review by hawkingrei — APPROVED** [review]

File: util/sessionutil/session.go [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+func (s *Session) Reset() {
+	s.StmtRollback(context.Background(), false)
+}
+
+// Execute is used to run one query internal.
+func (s *Session) Execute(ctx context.Context, query string, label string) ([]chunk.Row, error) {
+	startTime := time.Now()
+	var err error
+	defer func() {
+		metrics.DDLJobTableDuration.WithLabelValues(label + "-" + metrics.RetLabel(err)).Observe(time.Since(startTime).Seconds())
```
Comment (xhebox) [inline util/sessionutil/session.go:78]: Maybe it should be a submodule of ddl, since it is always using DDL source type and metrics?

Comment (Benjamin2037) [reply util/sessionutil/session.go:78]: No， it will be a common module, I will see how to deal with this.

**Review by wjhuang2016 — COMMENTED** [review]
Why do you want to make the session be a common function?

**Comment by Benjamin2037** [comment]
keep it as it is， refactor interface later

**Pull request closed without merging**